File: find.py

# This function finds the second smallest & second largest numbers
# from an array, and return the results in an array.


def find(list):

    print('inputs', list)

    if len(list) < 2:
        return "Invalid array"

    unique_sorted = []
    # remove dupliates
    for i in list:
        if i not in unique_sorted:
            unique_sorted.append(i)
    # sort it
    unique_sorted.sort()
    print('unique & sorted', unique_sorted)

    second_largest = unique_sorted[len(unique_sorted) - 2]
    second_smallest = unique_sorted[1]

    return [second_smallest, second_largest]


arr = [2, 24, 4, 10, 20, 5, 2, 20, 4]
print(find(arr))
arr = [100]
print(find(arr))
arr = []
print(find(arr))
arr = [2, 4]
print(find(arr))

File: find.js

// Find second smallest and second largest numbers
// from an array
function find(arr) {
  console.log('input:', arr);

  // validate
  if (arr.length < 2) {
    console.log("Invalid array");
  }

  // unique
  const unique = [...new Set(arr)];
  // sort
  unique.sort((a, b) => { return a - b });
  console.log('unique & sorted:', unique);

  // find secondSmallest and secondLargest
  const secondSmallest = unique[1];
  const secondLargest = unique[unique.length - 2];

  // Print
  console.log('secondSmallest', secondSmallest);
  console.log('secondLargest', secondLargest);
}

const items = [4, 5, 4, 6, 3, 4, 5, 2, 23, 1, 4, 4, 4];

find(items);